### ISSUE TITLE
Datagrid extension: Improve affordances for keyboard use in customize columns

### ISSUE THREAD

**Issue by elycheea** [carbon-design-system/ibm-products#2475]
## What package(s) are you using?

- [x] Carbon for IBM Products (or Carbon for IBM Cloud & Cognitive)
- [ ] Legacy/CDAI
- [ ] Legacy/Security

## Detailed description

Feedback from Mike Gower from Accessiblity —

> That is a slightly challenging implementation, to me, where when the whole ‘row’ has focus, you move it, and then you tab and only the checkboxes have focus. it’s a bit odd
> I think it would really help to have a cue that when the row has focus, you use the arrow keys to move it. Maybe if some kind of icon appeared? (say, what appears with sort?)

![Screen Recording 2022-11-14 at 9 07 57 AM](https://user-images.githubusercontent.com/1717398/202333389-2e1638fb-d210-47df-a4ba-a9e459bc766a.gif)

![image](https://user-images.githubusercontent.com/1717398/202333407-cdba54f7-aa28-4295-a576-f2fc1a435068.png)

> Describe in detail the issue you're having.

The selected state of selected rows when using keyboard actions does not make clear that we should use arrow keys to reorder items.

![image](https://user-images.githubusercontent.com/1717398/202332803-2d891d88-4aa1-49a1-9b29-58193752bd0b.png)

> Is this issue related to a specific component?

Datagrid extension for customizing columns

> What did you expect to happen? What happened instead? What would you like to
> see changed?

Additional affordances may aid users in reorganizing columns via keyboard

> What browser are you working in?

> What version of the @carbon/ibm-products (or @carbon/ibm-cloud-cognitive)
> package are you using?

1.35.2

> What offering/product do you work on? Any pressing ship or release dates we
> should be aware of?

## Steps to reproduce the issue

1. Navigate to datagrid > customize columns story in Storybook
2. Open the customize columns modal via mouse or keyboard
3. Use arrow keys to navigate rows then press space to select a row
4. From here, you can use arrow keys to reorder, but that may not be sufficiently obvious to users

> Please create a reduced test case in CodeSandbox
>
> https://codesandbox.io/s/github/carbon-design-system/ibm-cloud-cognitive/tree/main/examples/cloud-cognitive/codesandbox

## Additional information

- Screenshots or code
- Notes

**Comment by elycheea** [comment]
Needs design input. Should be brought up with designers after initial release.

**Comment by wkeese** [comment]
Presumably, focus should go to the drag handle, rather than the row, like done in https://schne324.github.io/dragon-drop/demo/.

I think https://www.smashingmagazine.com/2018/01/dragon-drop-accessible-list-reordering/ is a good summary of how to do this accessibly.  We're not too far off from his recommendations.

I know that Datagrid is deprecated but I'll look at making a PR for this.
